src/opencl-ecc-ed25519/main.cpp

```diff
@@ -0,0 +1,310 @@
+#include "cl_common.h"
+
+#include <stdio.h>
+#include <string.h>
+#include "ed25519.h"
+#include <inttypes.h>
```

Comment:
It would be nice to share this with the cuda version, only linking against a different dynamic library.

Reply:
But we don't need to block this PR for that.

Reply:
There are some differences between the OpenCL and CUDA variant so this would imply I would need to modify the CUDA variant. I wanted to avoid changing the CUDA files. A better integration can of course be done, but yes as you mentioned, I would be in favor of getting this merged "as is", without intervention on CUDA files. A future commit can make the integration better and further reduce duplication.